url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/pl/cc/SystemCoreState.java

```diff
@@ -0,0 +1,685 @@
+							RealQueue realQueue = findQueue(queue);
+							realQueue.updateAgentCounts();
+						} catch (QueueNotFoundException e) {
+							e.printStackTrace();
+						}
+					}
+				}
+			} catch (AgentNotFoundException e) {
+				log.error(e.getMessage());
+			}
+		} else if (cmd instanceof EventAgentQueueMember){
+			log.debug(cmd.getClass().getSimpleName()+" | "+cmd.getOrginalLine());
+			EventAgentQueueMember e = (EventAgentQueueMember)cmd;
+			try {
+				/*
+				 * FIXME: marek wysyła extension zamiast id
+				 * Docelowo zdecydować sie czy przesyłamy id czy extension
+				 * RealAgent ra = findAgent(e.getAgent(),e.getQueue());
+				 */
+				RealAgent ra;
+				ra = (RealAgent) aList.getByExtension(e.getAgent().getId().toString());
+				if (ra==null){
+					log.debug("Nie znaleziono agenta (szukając po ext): "+e.getAgent().getId());
+					ra = aList.get(e.getAgent());
+				}
+				if (ra == null){
+					log.warn("Nie znaleziono agenta o id/exten: "+e.getAgent().getId());
+					return false;
+				}
+				AgentStatusInQueue status = e.getStatus();
+
+				try {
+					// Zwyky agent nie będący supervisorem nie otrzymuje informacji o kolejkach
+					// FIXME: docelowo powinien te informacje otrzymać
+					ra = findAgent(ra, e.getQueue());
+					log.debug("EventAgentQueueMember - TRANSLATED agentId: "+ra.getId());
+					ra.onChangeStatusAsQueueMember(e.getQueue(), status);
+					
+				} catch (AgentNotFoundException e1) {
+				} catch (QueueNotFoundException e2) {
+				}
+				// lista zalogowanych agentów 
+				RealAgent rootAgent = findAgent(ra);
+				
+				rootAgent.onChangeStatusAsQueueMember(e.getQueue(), status);
+				
+				if (status==AgentStatusInQueue.NOT_INUSE){
+					RealCall call = new RealCall("","123");
+					rootAgent.onHangeup(call);
+				}
+				
+				// ilość wolnych
+				RealQueue realQueue = findQueue(e.getQueue());
+				realQueue.updateAgentCounts();
+			} catch (AgentNotFoundException e1) {
+				log.error(e1.getMessage());
+			} catch (QueueNotFoundException e2) {
+				log.error(e2.getMessage());
+			}
+			
+		} else if (cmd instanceof EventQueueJoin){
+			// połączenie wpadło do kolejki
+			log.debug(cmd.getClass().getSimpleName()+" | "+cmd.getOrginalLine());
+			RealCall c = ((EventQueueJoin)cmd).getCall();
+			RealQueue q = ((EventQueueJoin)cmd).getQueue();
+			addCallToList(c, q);
+			
+		} else if (cmd instanceof InfoPrivilege){
+			RealAgent agent = ((InfoPrivilege)cmd).getAgent();
+			boolean hasPriv = ((InfoPrivilege)cmd).getPermission();
+			if (hasPriv){
+				privAgentList.add(agent);
+			}
+		} else if (cmd instanceof InfoPauseDefinition){
+			PauseType pauseType = ((InfoPauseDefinition)cmd).getPauseType();
+			pauseTypeList.add(pauseType);
+		} else {	
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Wspólna obsługa zdażeń EventAgentComplete i EventAgentHangup
+	 * @param agent2
+	 * @param call2
+	 */
+	private void callCompleded(final RealAgent agent2, final RealCall call2 ){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 51 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 410-489 into a separate method.